docs/content/guides/developer/advanced/move-2024-migration.mdx

```diff
@@ -0,0 +1,300 @@
+
+## How to migrate
+
+To migrate a project to Move 2024 Beta, you have two options:
+
+1. Delete your existing `Move.lock` file (if one exists) to make sure you're using the newest `sui-framework` version.
+2. Run `sui move migrate` in the root of your Move project. See [Automatic migration](#automatic-migration). 
+   1. Alternatively, update your `Move.toml` file's `[package]` entry to include `edition = "2024.beta"`. If you do this, you might receive a number of new errors as part of our [breaking changes](#breaking-changes).
+
+## Automatic migration {#automatic-migration}
+
+Move 2024 includes an automatic migration script that you can use by calling `sui move migrate` in the root of your Move project. Upon running, your console prompts you for which Move edition to use. If you select `2024.beta`, the script invokes the compiler and attempts to automatically update your code to avoid the [breaking changes](#breaking-changes) the update introduces (including marking structs as `public`, mutable variables with the `mut` keyword, avoiding restricted keywords, swapping `friend`s for `public(package)`, and even updating paths to global paths in many cases).
+
+After this is done, your console displays a diff of the changes the script intends to make. If you accept the changes, the script updates your code and your `Move.toml` file automatically. You are now using Move 2024 Beta.
+
+## New features
+
+Here is a brief overview of some of the new features in Move 2024.
+
+### Method syntax
+
+You can call certain functions now as methods using the `.` syntax. For example, the following call
+
+```move
+vector::push_back(&mut v, coin::value(&c));
+```
+
+can now be written as
+
+```move
+v.push_back(c.value());
+```
+
+Where the receiver of the method (`v` and `c` in this example) is automatically borrowed if necessary (as `&mut v` and `&c` respectively).
+
+You can call any function defined in the same module as the receiver's type as a method if it takes the receiver as its first argument.
+
+For functions defined outside the module, you can declare methods using `public use fun` and `use fun`.
+
+### Index syntax
+
+With method syntax, you can annotate certain functions as being `#[syntax(index)]` methods. You then call these methods using `v[i]`-style calls.
+
+For example,
+
+```move
+*&mut v[i] = v[j];
+```
+
+resolves to
+
+```move
+*vector::borrow_mut(&mut v, i) = *vector::borrow(&v, j);
+```
+
+### public(package) {#public-package}
+
+`friend` declarations, and the associated `public(friend)` visibility modifiers, are deprecated. In their place is the `public(package)` visibility modifier, which allows calling functions only within the same package where they are defined.
+
+### Positional fields
+
+You can now define `struct`s with positional fields, which are accessed by zero-based index. For example,
+
+```move
+public struct Pair(u64, u64) has copy, drop, store.
```

Comment:
I believe that period shouldn't be there?

Reply:
Semi-colon, but good catch!